opentelemetry-api/src/opentelemetry/trace/span.py

```diff
@@ -420,7 +420,11 @@ def __new__(
         if trace_state is None:
             trace_state = DEFAULT_TRACE_STATE
 
-        is_valid = trace_id != INVALID_TRACE_ID and span_id != INVALID_SPAN_ID
+        is_valid = (
+            trace_id != INVALID_TRACE_ID
+            and span_id != INVALID_SPAN_ID
+            and trace_id < 2 ** 128 - 1
```

Comment:
@ocelotl regarding the single use constant, I think it's worth having the constant as it's easier to understand what the magic number means and for the speedup of not calculating the value every time in this hot code path.

(I was curious so checked and CPython is not smart enough to optimize this into a constant on its own (funny enough, it does do it for the bit shifting approach)):
```ipynb
In [2]: def f(trace_id):
   ...:     return trace_id < 2 ** 128 - 1
   ...:

In [3]: dis(f)
  2           0 LOAD_FAST                0 (trace_id)
              2 LOAD_CONST               1 (2)
              4 LOAD_CONST               2 (128)
              6 BINARY_POWER
              8 LOAD_CONST               3 (1)
             10 BINARY_SUBTRACT
             12 COMPARE_OP               0 (<)
             14 RETURN_VALUE
```

Reply:
Ok, that's a good point. In that case, the constant should be added as a private attribute of the class to keep it as close as where it is being used.

Reply:
Not sure if we have specs somewhere, but my 2 cents is that I think `(1 << 128) - 1` is easier to read.

I can go either way on the constant though!

Reply:
ok I put the private const variable for readabiliy. As per bit shift vs multiplication, I leave the decision.